x/evm/types/hooks.go

```diff
@@ -0,0 +1,96 @@
+package types
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+var (
+	// BankSendEvent represent the signature of
+	// `event __CosmosNativeBankSend(address recipient, uint256 amount, string denom)`
+	BankSendEvent abi.Event
+
+	_ EvmHooks = MultiEvmHooks{}
+	_ EvmHooks = BankSendHook{}
+)
+
+// MultiEvmHooks combine multiple governance hooks, all hook functions are run in array sequence
+type MultiEvmHooks []EvmHooks
+
+// NewMultiEvmHooks combine multiple governance hooks
+func NewMultiEvmHooks(hooks ...EvmHooks) MultiEvmHooks {
+	return hooks
+}
+
+// PostTxProcessing delegate the call to underlying hooks
+func (h MultiEvmHooks) PostTxProcessing(ctx sdk.Context, txHash ethcmn.Hash, logs []*ethtypes.Log) error {
+	for i := range h {
+		err := h[i].PostTxProcessing(ctx, txHash, logs)
+		if err != nil {
+			return err
```

Comment:
```suggestion
	for i := range mh {
		if err := mh[i].hook.PostTxProcessing(ctx, txHash, logs); err != nil {
			return sdkerrors.Wrapf(err, "module %s EVM hooks failed", mh[i].module)
```

Reply:
rather than add a new method to the interface, how about just report the reflected type name:
```
return sdkerrors.Wrapf(err, "EVM hook %s failed", reflect.TypeOf(mh[i]))
```

Reply:
done

Reply:
let's use `%T` instead of `reflect`

Reply:
done